core/domain/prod_validation_jobs_one_off.py

```diff
@@ -0,0 +1,133 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for validating prod models."""
+import logging
+
+from core import jobs
+from core.platform import models
+
+
+(
+    user_models, exp_models, collection_models,
+    feedback_models) = models.Registry.import_models([
+        models.NAMES.user, models.NAMES.exploration,
+        models.NAMES.collection, models.NAMES.feedback])
+
+INVALID_USER_SUBSCRIPTION_MODEL = 'invalid_user_subscription_model'
+VALID_USER_SUBSCRIPTION_MODEL = 'valid_user_subscription_model'
+
+
+
+def has_model_for_id(model_class, model_id):
+    """Checks there's an existing nondeleted model for given model_id.
+
+    Args:
+        model_class: BaseModel. Class of the model.
+        model_id: str. Key of the model.
+
+    Returns:
+        bool. True if there exists a model with given model_id.
+        """
+    model = model_class.get_by_id(model_id)
+    return model is not None and model.deleted is False
+
+
+class ValidateUserSubscriptionsModel(object):
+    """Validate UserSubscriptionsModels."""
+
+    errors = []
+    checks_passed = 0
+
+    @classmethod
+    def validate(cls, item):
+        """Validates a UserSubscriptionsModel entity.
+
+        Args:
+            item: UserSubscriptionsModel. Entity to validate.
+
+        Returns:
+            string. Validation status.
+        """
+        external_ids_checks = {
+            'activity_ids': exp_models.ExplorationModel,
+            'collection_ids': collection_models.CollectionModel,
+            'general_feedback_thread_ids':
+            feedback_models.GeneralFeedbackThreadModel,
+            'creator_ids': user_models.UserSettingsModel,
+        }
+        if not has_model_for_id(user_models.UserSettingsModel, item.id):
```

Comment:
Take a look at plaform/datastore/gae_datastore_services. It is probably more efficient to compile all the models and IDs you need, then use that function to fetch them all at once. 

Then have a streamlined way to convert missing models into useful error messages. The error message should specify: the ID/type of the current primary model being looked into, and something like "based on field X having value Y, expected other model with type Z, id W to be found, but it didn't exist". That should give enough info to debug.

Reply:
I think it gives the corresponding entry being None for missing models? 

If not, might want to modify it to make it so (maybe introduce strict=True and strict=False modes -- the latter means None is returned in the place of nonexistent entities, and the former means an exception is raised if any entity doesn't exist). But if so, then I think the existing behaviour works for your use case and no changes are needed.

Reply:
It does look like it returns None for missing models!